utils/UUID_gen.hh

```diff
@@ -338,7 +338,7 @@ public:
      * @param uuid
      * @return milliseconds since Unix epoch
      */
-    static decimicroseconds normalized_timestamp(UUID uuid)
+    static decimicroseconds normalized_timestamp(const UUID& uuid)
```

Comment:
It's probably slower even without the optimizer. The uuid can be passed in two registers, instead of a single register + two memory loads.